Integration test for tsconfig fails locally

### What happened?

When I ran integration test for tsconfig locally, the test failed.
It expected `@code-pushup/models` not to be found but instead they were found.

### What would you expect to happen?

The test is implemented in a way that it passes in all supported environments.

### What steps did you take?

I ran the test locally using `npx nx run-many -t integration-test`

### Code PushUp package version

0.26.1

### What operation system are you on?

Windows

### Node version

18.18.2

### Relevant log output

```shell
AssertionError: promise resolved "{ …(7) }" instead of rejecting

- Expected
+ Received

- [Error: rejected promise]
+ Object {
+   "categories": Array [],
+   "config": "C:\\Users\\P\\WebstormProjects\\qm-cli\\testing\\test-utils\\src\\lib\\fixtures\\configs\\code-pushup.needs-tsconfig.config.ts",
+   "onlyPlugins": Array [],
+   "persist": Object {
+     "filename": "report",
+     "format": Array [
+       "json",
+     ],
+     "outputDir": ".code-pushup",
+   },
+   "plugins": Array [
+     Object {
+       "audits": Array [
+         Object {
+           "slug": "always-perfect",
+           "title": "Always perfect",
+         },
+       ],
+       "icon": "javascript",
+       "runner": [Function anonymous],
+       "slug": "good-feels",
+       "title": "Good feels",
+     },
+   ],
+   "progress": true,
+   "verbose": false,
+ }
```

--- a/packages/cli/src/lib/implementation/core-config.middleware.integration.test.ts
+++ b/packages/cli/src/lib/implementation/core-config.middleware.integration.test.ts
@@ -47,7 +47,7 @@ describe('coreConfigMiddleware', () => {
     await expect(
       coreConfigMiddleware({
         config: join(configDirPath, 'code-pushup.needs-tsconfig.config.ts'),
-        tsconfig: 'tsconfig.base.json',
+        tsconfig: join(configDirPath, 'tsconfig.json'),
         ...CLI_DEFAULTS,
       }),
     ).resolves.toBeTruthy();
@@ -59,6 +59,6 @@ describe('coreConfigMiddleware', () => {
         config: join(configDirPath, 'code-pushup.needs-tsconfig.config.ts'),
         ...CLI_DEFAULTS,
       }),
-    ).rejects.toThrow("Cannot find package '@code-pushup/models'");
+    ).rejects.toThrow("Cannot find package '@example/custom-plugin'");
   });
 });

--- a/packages/core/src/lib/implementation/read-rc-file.integration.test.ts
+++ b/packages/core/src/lib/implementation/read-rc-file.integration.test.ts
@@ -41,7 +41,7 @@ describe('readRcByPath', () => {
     await expect(
       readRcByPath(
         join(configDirPath, 'code-pushup.needs-tsconfig.config.ts'),
-        'tsconfig.base.json',
+        join(configDirPath, 'tsconfig.json'),
       ),
     ).resolves.toEqual({
       plugins: [

--- a/testing/test-utils/.eslintrc.json
+++ b/testing/test-utils/.eslintrc.json
@@ -1,6 +1,6 @@
 {
   "extends": ["../../.eslintrc.json"],
-  "ignorePatterns": ["!**/*"],
+  "ignorePatterns": ["!**/*", "src/lib/fixtures/configs"],
   "overrides": [
     {
       "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],

--- a/testing/test-utils/src/lib/fixtures/configs/code-pushup.needs-tsconfig.config.ts
+++ b/testing/test-utils/src/lib/fixtures/configs/code-pushup.needs-tsconfig.config.ts
@@ -1,34 +1,9 @@
-// the point is to test runtime import which relies on alias defined in tsconfig "paths"
-// importing schemas from '@code-pushup/models' wouldn't work without --tsconfig
-import {
-  type Audit,
-  type AuditOutputs,
-  type CoreConfig,
-  auditOutputsSchema,
-  auditSchema,
-} from '@code-pushup/models';
+// the point is to test runtime import which relies on alias defined in tsconfig.json "paths"
+// non-type import from '@example/custom-plugin' wouldn't work without --tsconfig
+import customPlugin from '@example/custom-plugin';
 
-export default {
-  plugins: [
-    {
-      slug: 'good-feels',
-      title: 'Good feels',
-      icon: 'javascript',
-      audits: [
-        auditSchema.parse({
-          slug: 'always-perfect',
-          title: 'Always perfect',
-        } satisfies Audit),
-      ],
-      runner: () =>
-        auditOutputsSchema.parseAsync([
-          {
-            slug: 'always-perfect',
-            score: 1,
-            value: 100,
-            displayValue: '✅ Perfect! 👌',
-          },
-        ] satisfies AuditOutputs),
-    },
-  ],
-} satisfies CoreConfig;
+const config = {
+  plugins: [customPlugin],
+};
+
+export default config;

--- a/testing/test-utils/src/lib/fixtures/configs/custom-plugin.ts
+++ b/testing/test-utils/src/lib/fixtures/configs/custom-plugin.ts
@@ -1,0 +1,21 @@
+const customPlugin = {
+  slug: 'good-feels',
+  title: 'Good feels',
+  icon: 'javascript',
+  audits: [
+    {
+      slug: 'always-perfect',
+      title: 'Always perfect',
+    },
+  ],
+  runner: () => [
+    {
+      slug: 'always-perfect',
+      score: 1,
+      value: 100,
+      displayValue: '✅ Perfect! 👌',
+    },
+  ],
+};
+
+export default customPlugin;

--- a/testing/test-utils/src/lib/fixtures/configs/tsconfig.json
+++ b/testing/test-utils/src/lib/fixtures/configs/tsconfig.json
@@ -1,0 +1,7 @@
+{
+  "compilerOptions": {
+    "paths": {
+      "@example/custom-plugin": ["./custom-plugin.ts"]
+    }
+  }
+}

--- a/testing/test-utils/tsconfig.lib.json
+++ b/testing/test-utils/tsconfig.lib.json
@@ -6,5 +6,9 @@
     "types": ["node"]
   },
   "include": ["src/**/*.ts"],
-  "exclude": ["vite.config.unit.ts", "src/**/*.unit.test.ts"]
+  "exclude": [
+    "vite.config.unit.ts",
+    "src/**/*.unit.test.ts",
+    "src/lib/fixtures/configs"
+  ]
 }

